### PR TITLE
feature: keep prediction in UI

### PR DESCRIPTION
**Pull request by alex-krasn**
We're keeping predictionPage mounted until switching to another project, then remount it.

### CODE REVIEW

File: src/react/components/shell/mainContentRouter.tsx [hunk 1]
```diff
@@ -15,7 +16,27 @@ import ProjectSettingsPage from "../pages/projectSettings/projectSettingsPage";
  * @name - Main Content Router
  * @description - Controls main content pane based on route
  */
+
+interface IProjectId {
+    projectId: string;
+}
+
 export function MainContentRouter() {
+    const id = getPathParams(window.location.pathname, "projectId") as IProjectId;
```
Comment (stew-ro) [inline src/react/components/shell/mainContentRouter.tsx:25]: cant we get the id from the Redux store instead of the url

Comment (alex-krasn) [reply src/react/components/shell/mainContentRouter.tsx:25]: I think, it is faster to get it from path directly neither wait for response from Redux. Also Redux reload many times... 
image

Comment (stew-ro) [reply src/react/components/shell/mainContentRouter.tsx:25]: it seems redundant to add a projectId state to this component when it's already in the redux store. Usually you want to lift state. But maybe not necessary in this case

File: src/common/currPathParams.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { matchPath } from "react-router-dom";
+
+
+export const getPathParams = (path: string, param: string) => {
```
Comment (stew-ro) [inline src/common/currPathParams.ts:4]: we're no longer using this right? We're getting from redux store instead. Probably can remove

Comment (alex-krasn) [reply src/common/currPathParams.ts:4]: I though I deleted the file...let me fixit

File: src/react/components/shell/preditcPageRoute.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+export function PredictPageRoute() {
+    const projectProperties = useSelector((state: IApplicationState) => {
+        if (state && state.currentProject) {
+            const { apiKey, folderPath, apiUriBase, id, trainRecord } = state.currentProject;
+            let modelId: string;
+            if (trainRecord) {
+                modelId = trainRecord.modelInfo.modelId;
+            }
+            return JSON.stringify({ id, apiKey, apiUriBase, folderPath, modelId });
```
Comment (stew-ro) [inline src/react/components/shell/preditcPageRoute.tsx:25]: could leave out apiKey and apiUriBase I think

Comment (alex-krasn) [reply src/react/components/shell/preditcPageRoute.tsx:25]: I thought if they will switch, lets say, different version on the service - result can varies so we should unmount prediction...

**Pull request merged**